src/xshell_base.hpp

```diff
@@ -21,25 +21,25 @@ namespace xeus
 {
     class xserver_zmq_split;
 
-    class xshell
+    class xshell_base
     {
     public:
 
         using listener = std::function<void(xmessage)>;
 
-        xshell(zmq::context_t& context,
+        xshell_base(zmq::context_t& context,
```

Comment:
This class should implement proper entity semantics:
- the constructor should be moved to a protected section
- copy and move semantics should be deleted in the public section.